### ISSUE TITLE
Use url::Url or allow passing strings to `Redirect`

### ISSUE THREAD

**Issue by JoelEllis** [poem-web/poem#176]
## Description of the feature

It would be ideal if Poem did not use its own `Uri` and insead used the [`url` crate](https://docs.rs/url/latest/url/) for type-interoperability with other crates, like `reqwest`, etc. 

## Code example (if possible)

At the moment I have to do this:

```rust
    // Takes `poem::Uri`, results in URL being validated needlessly
    Ok(Redirect::temporary(poem::http::Uri::from_str(redirect.as_str())?))
```

I would prefer:

```rust
    // Takes `Url` type  
    Ok(Redirect::temporary(redirect))

    // Takes `String` type  
    Ok(Redirect::temporary(redirect.to_string()))
```

**Comment by sunli829** [comment]
You are right, so now the type is changed from `Uri` to `impl Display`.

**Comment by sunli829** [comment]
Released in `v1.2.44`

**Comment by JoelEllis** [comment]
Thank you!
